### PR TITLE
Add `parse()` and `tryparse()` for `SHA1` and `UUID`

### PR DESCRIPTION
**Pull request by staticfloat**
Inspired by Kristoffer's comments in https://github.com/JuliaLang/julia/pull/37320/files

### CODE REVIEW

File: base/loading.jl [hunk 1]
```diff
@@ -91,6 +91,17 @@ struct SHA1
     end
 end
 SHA1(s::AbstractString) = SHA1(hex2bytes(s))
+parse(::Type{SHA1}, s::AbstractString) = SHA1(s)
+function tryparse(::Type{SHA1}, s::AbstractString)
+    try
+        return parse(SHA1, s)
+    catch e
+        if isa(e, InterruptException)
```
Comment (stevengj) [inline base/loading.jl:99]: Wouldn't it be better to specifically return `nothing` for an `ArgumentError` and `rethrow` everything else?

Comment (staticfloat) [reply base/loading.jl:99]: Yes, that does seem better to me.

**Comment by KristofferC** [comment]
This is the wrong way around (in my opinion). There should not be any throws in the code when you do a `tryparse`. `tryparse` is the inner one and `parse` calls `tryparse` and does all the throwing depending on the return value of `tryparse`.

**Review by KristofferC — APPROVED** [review]
The stuff in my comment can be tweaked in a follow up PR though, now when the API is in place.

**Comment by staticfloat** [comment]
I do think it's important to continue to get the rich information from the `parse()` function failing.  I don't think swallowing errors in `tryparse()` is that big of a deal; the contract of `tryparse()` seems like it's valid for it to swallow all errors greedily.  It would be nice if we can narrow down the source of errors, (As done in this PR to only filter out `ArgumentError`s) but if I am to `tryparse(Foo, x)` I really don't ever want that to throw for any reason that is attributable to `x`.

**Pull request merged**